opencti-platform/opencti-front/src/private/components/common/location/LocationCreation.tsx

```diff
@@ -0,0 +1,277 @@
+  description: Yup.string().nullable(),
+  type: Yup.string().required(t('This field is required')),
+});
+
+const LocationCreationForm: FunctionComponent<LocationCreationFormProps> = ({
+  inputValue,
+  onlyAuthors,
+  onReset,
+  onCompleted,
+  contextual,
+  creationCallback,
+  updater,
+}) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const [commit] = useMutation<LocationCreationMutation>(locationMutation);
+
+  const onSubmit: FormikConfig<LocationAddInput>['onSubmit'] = (
+    values,
+    {
+      setSubmitting,
+      resetForm,
+      setErrors,
+    },
+  ) => {
+    const input: LocationCreationMutation$variables['input'] = {
+      name: values.name,
+      description: values.description,
+      type: values.type,
+    };
+    commit({
+      variables: {
+        input,
+      },
+      updater: (store) => updater(store),
+      onError: (error: Error) => {
+        handleErrorInForm(error, setErrors);
+        setSubmitting(false);
+      },
+      onCompleted: (response) => {
+        setSubmitting(false);
+        resetForm();
+        if (contextual && creationCallback) {
+          creationCallback(response);
+        }
+        if (onCompleted) {
+          onCompleted();
+        }
+      },
+    });
+  };
+
+  return (
+    <Formik
+      enableReinitialize={true}
+      initialValues={{
+        name: inputValue,
+        description: '',
+        type: '',
+      }}
+      validationSchema={locationValidation(t)}
+      onSubmit={onSubmit}
+      onReset={onReset}
+    >
+      {({
+        submitForm,
+        handleReset,
+        isSubmitting,
+      }) => (
+        <Form style={{ margin: '20px 0 20px 0' }}>
+          <Field
+            component={TextField}
+            variant="standard"
+            name="name"
+            label={t('Name')}
+            fullWidth={true}
+            detectDuplicate={['Organization', 'Individual']}
+          />
+          <Field
+            component={MarkdownField}
+            name="description"
+            label={t('Description')}
+            fullWidth={true}
+            multiline={true}
+            rows="4"
+            style={{ marginTop: 20 }}
+          />
+          <Field
+            component={SelectField}
+            variant="standard"
+            name="type"
+            label={t('Entity type')}
+            fullWidth={true}
+            containerstyle={fieldSpacingContainerStyle}
+          >
+            {!onlyAuthors && (
+            <MenuItem value="Administrative-Area">{t('Administrative Area')}</MenuItem>
+            )}
+            {!onlyAuthors && (
+              <MenuItem value="Region">{t('Region')}</MenuItem>
+            )}
+            {!onlyAuthors && (
+              <MenuItem value="Country">{t('Country')}</MenuItem>
+            )}
+            {!onlyAuthors && (
+              <MenuItem value="City">{t('City')}</MenuItem>
+            )}
+            <MenuItem value="Position">{t('Position')}</MenuItem>
```

Comment:
why is there no !onlyAuthors test for Position?

Reply:
Because I forgot about it ^^